Evaluate lookup terms only in with_* contexts

janLo opened the pull request:
As mentioned in Issue #10073 there is a problem when evaluation lookup
terms in inline {{ lookup( ... ) }} statements. The problem is, that
the template engine dies not know that the term was a literal even if
it was quoted, because the "outer" tempalte engine have already
stripped the quotes when the templating in listify_lookup_plugin_terms
see the term.

The term will then be evaluated as template variables and if a part
of the term evaluates to the "parent" variable of the current scope it
will cause an endless recursion.

This patch fixes the issue by introducing a new inject property that
tells in whick context we are currently. So listify_lookup_plugin_terms
can skip term evaluation if we are not within a "with_*" context.

janLo committed:
Evaluate lookup terms only in with_* contexts

As mentioned in Issue #10073 there is a problem when evaluation lookup
terms in inline {{ lookup( ... ) }} statements. The problem is, that
the template engine dies not know that the term was a literal even if
it was quoted, because the "outer" tempalte engine have already
stripped the quotes when the templating in listify_lookup_plugin_terms
see the term.

The term will then be evaluated as template variables and if a part
of the term evaluates to the "parent" variable of the current scope it
will cause an endless recursion.

This patch fixes the issue by introducing a new inject property that
tells in whick context we are currently. So listify_lookup_plugin_terms
can skip term evaluation if we are not within a "with_*" context.

abadger commented:
I believe this is fixed in v2.  Still evaluating whether this sohuld be applied to 1.9.x as well.  May be too big of a change.

janLo committed:
Return a list from listify_lookup_plugin_terms

The quirk to not evaluate literals from the lookup definition
(issue #10073 upstream) breaks the password lookup because it require
listify_lookup_plugin_terms to return a list.

This commit fixes this by add a check if we try to return a basestring
and put it in a list if this is the case.

jimi-c commented:
Per the comment on the other issue, since this has been applied to the devel branch we're going to go ahead and close this now.

Thanks!
